sdk/iot/provisioning/src/az_iot_provisioning_client_sas.c

```diff
@@ -142,17 +128,24 @@ AZ_NODISCARD az_result az_iot_provisioning_client_sas_get_password(
   if (az_span_size(key_name) > 0)
   {
     // Key Name
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(mqtt_password_span,
+      1 // AMPERSAND
```

Comment:
nit: I prefer having each of these `1`'s as a call to `sizeof()` but not a blocker.

Reply:
I tried using sizeof but had some issues. Will address that in separate.